### DISCUSSION TITLE
Replace html anchors with React Router Link component

### DISCUSSION THREAD

**Discussion by romatou** [sibiraj-s/marked-react#20] [Q&A] [answered]
I was implementing markdown parser in my project and ran into a problem where links with relative paths were redirecting to a page and clearing state (acting as standard anchors) due to page reload. React Router Link component or useNavigate() secures the page reload and keeps the state. I was [suggested](https://github.com/markedjs/marked/discussions/2982#discussioncomment-7121553) to use marked-react for this but it didn't solve my problem. 

**Could you please provide solution for replacing anchors with Link component?**

**Comment by sibiraj-s** [comment]
can you try custom renderer. Example can be found here.

https://github.com/sibiraj-s/marked-react#syntax-highlight-code-blocks

I will add the custom renderer to the Readme as a separate block.

**Reply by sibiraj-s** [reply, marked as answer]
It will be like this

```js
const renderer = {
  link(snippet, lang) {
    return <Link />;
  },
};

<Markdown renderer={renderer} />
```
